src/hooks/loadables/useLoadRecovery.ts

```diff
@@ -0,0 +1,98 @@
+import { SENTINEL_ADDRESS } from '@safe-global/safe-core-sdk/dist/src/utils/constants'
+import { BigNumber } from 'ethers'
+import type { Delay } from '@gnosis.pm/zodiac'
+
+import { getDelayModifiers } from '@/services/recovery/delay-modifier'
+import useAsync from '../useAsync'
+import useSafeInfo from '../useSafeInfo'
+import { useWeb3ReadOnly } from '../wallets/web3'
+import { getSpendingLimitModuleAddress } from '@/services/contracts/spendingLimitContracts'
+import useIntervalCounter from '../useIntervalCounter'
+import { useHasFeature } from '../useChains'
+import { FEATURES } from '@/utils/chains'
+import type { AsyncResult } from '../useAsync'
+import type { RecoveryState } from '@/store/recoverySlice'
+
+const MAX_PAGE_SIZE = 100
+const REFRESH_DELAY = 5 * 60 * 1_000 // 5 minutes
+
+const getRecoveryState = async (delayModifier: Delay): Promise<RecoveryState[number]> => {
+  const transactionAddedFilter = delayModifier.filters.TransactionAdded()
+
+  const [[modules], txExpiration, txCooldown, txNonce, queueNonce, transactionsAdded] = await Promise.all([
+    delayModifier.getModulesPaginated(SENTINEL_ADDRESS, MAX_PAGE_SIZE),
+    delayModifier.txExpiration(),
+    delayModifier.txCooldown(),
+    delayModifier.txNonce(),
+    delayModifier.queueNonce(),
+    delayModifier.queryFilter(transactionAddedFilter),
+  ])
+
+  const queue = await Promise.all(
+    transactionsAdded
+      // Only queued transactions with queueNonce >= current txNonce
+      .filter(({ args }) => args.queueNonce.gte(txNonce))
+      .map(async (event) => {
+        const txBlock = await event.getBlock()
+
+        const validFrom = BigNumber.from(txBlock.timestamp).add(txCooldown)
+        const expiresAt = txExpiration.isZero()
+          ? null // Never expires
+          : validFrom.add(txExpiration)
+
+        return {
+          ...event,
+          timestamp: txBlock.timestamp,
+          validFrom,
+          expiresAt,
+        }
+      }),
+  )
```

Comment:
I would extract this into its own function.

Reply:
I've extracted it and added test coverage for it (and all other loading related logic) in 181b21d9cfe796e6960f6ba52f531759e34f7184.